cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -89,6 +90,77 @@ public GradebookDTO getGradebook(@PathVariable("id") Integer assignmentId  ) {
 		return gradebook;
 	}
 	
+	@PostMapping("/course/{course_id}/add")
+	public void addAssignment(@RequestBody AssignmentDTO assignmentDTO, @PathVariable int course_id) {
+		
+		// check that this request is from the course instructor 
+		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
+		
+		// grab requested course and validate permissions if it exists
+		Course c = courseRepository.findById(course_id).orElse(null);
+		
+		if (!c.getInstructor().equals(email)) {
+			throw new ResponseStatusException( HttpStatus.UNAUTHORIZED, "Not Authorized. " );
+		}
+		
+		// create new assignment
+		Assignment assignment = new Assignment();
+		assignment.setName(assignmentDTO.assignmentName);
+		assignment.setDueDate(java.sql.Date.valueOf(assignmentDTO.dueDate));
+		assignment.setCourse(c);
+		assignmentRepository.save(assignment);
+		
```

Comment:
The original code given for the registration addCourse method in the schedule controller returns the courseDTO. I am not sure if we have to follow that format, but you could return the assignmentDTO here.

Reply:
I think I'll add this if I need it when writing the front end portion of the assignment